data/abilities.js

```diff
@@ -840,11 +840,17 @@ exports.BattleAbilities = {
 	},
 	"emergencyexit": {
 		shortDesc: "This Pokemon switches out when it reaches 1/2 or less of its maximum HP.",
-		// TODO
 		id: "emergencyexit",
 		name: "Emergency Exit",
 		rating: 3,
 		num: 194,
+		onHit: function (target) {
+			if (!this.canSwitch(target.side) || target.forceSwitchFlag) return;
+			if (target.hp <= target.maxhp/2) {
+				target.switchFlag = true;
```

Comment:
I believe these Abilities block U-turn/Volt Switch when they activate, so you have to set the `switchFlag` to `false` on the `source` as well. https://www.smogon.com/forums/posts/7074504